### PR TITLE
Merge bitcoin#13152: Add getnodeaddresses RPC command

### PR DESCRIPTION
**Pull request by dzutte-cpp**
bitcoin#13152

### CODE REVIEW

File: test/functional/test_framework/test_framework.py [hunk 1]
```diff
@@ -104,6 +104,7 @@ def __init__(self):
         """Sets test framework defaults. Do not override this method. Instead, override the set_test_params() method"""
         self.chain = 'regtest'
         self.setup_clean_chain = False
+        self.mock_time_on_setup = True
```
Comment (dzutte-cpp) [inline test/functional/test_framework/test_framework.py:107]: @UdjinM6 could you please suggest if the proposed option is the right way to go here?
Before this change, Dash functional tests would always mock time, whereas original Bitcoin tests do it case-by-case.

Comment (UdjinM6) [reply test/functional/test_framework/test_framework.py:107]: That's might be an interesting approach but it should be implemented in a separate PR imo (and applied to other tests that override `setup_network` to use `disable_mocktime`). For this PR I would rather go with https://github.com/UdjinM6/dash/commit/fa4797b5bab60b89edc4a9943ee2bc4e9635153e

**Review by UdjinM6 — CHANGES_REQUESTED** [review]
see https://github.com/dashpay/dash/pull/4425#discussion_r708675811

**Review by UdjinM6 — APPROVED** [review]
LGTM, utACK

File: test/functional/rpc_net.py [hunk 2]
```diff
@@ -102,5 +111,40 @@ def _test_getpeerinfo(self):
...
+        msg = msg_addr()
+        imported_addrs = []
+        for i in range(256):
+            a = "123.123.123.{}".format(i)
+            imported_addrs.append(a)
+            addr = CAddress()
+            addr.time = 100000000
+            addr.nServices = NODE_NETWORK
+            addr.ip = a
+            addr.port = 8333
```
Comment (PastaPastaPasta) [inline test/functional/rpc_net.py:127]: This should probably be dashified right? @UdjinM6

File: test/functional/rpc_net.py [hunk 3]
```diff
@@ -102,5 +111,40 @@ def _test_getpeerinfo(self):
...
+
+        # obtain addresses via rpc call and check they were ones sent in before
+        REQUEST_COUNT = 10
+        node_addresses = self.nodes[0].getnodeaddresses(REQUEST_COUNT)
+        assert_equal(len(node_addresses), REQUEST_COUNT)
+        for a in node_addresses:
+            assert_greater_than(a["time"], 1527811200) # 1st June 2018
+            assert_equal(a["services"], NODE_NETWORK)
+            assert a["address"] in imported_addrs
+            assert_equal(a["port"], 8333)
```
Comment (PastaPastaPasta) [inline test/functional/rpc_net.py:139]: same

**Review by PastaPastaPasta — CHANGES_REQUESTED** [review]

File: test/functional/rpc_net.py (hunk 2 above)
Comment (UdjinM6) [reply test/functional/rpc_net.py:127]: Regular nodes can have ports other than the default one (unlike MNs), so technically you can run a node on port 8333 and it shouldn't matter here.

Comment (PastaPastaPasta) [reply test/functional/rpc_net.py:127]: Yeah, it doesn't REALLY matter, but feels better to use the default port, but I guess it doesn't really matter.

**Review by PastaPastaPasta — APPROVED** [review]
utACK for squash merge

up to @dzutte-cpp whether to change port or not.

**Comment by UdjinM6** [comment]
That's not the only place were we use non-default ports really. I'm just going to merge it as is, feel free to submit a PR that changes all the 8333 ports to 9999 ones :)

**Pull request merged**